### DISCUSSION TITLE
? Keyboard Help screen should reflect changed/deleted mappings.

### DISCUSSION THREAD

**Discussion by onguarde** [brookhong/Surfingkeys#1679] [Ideas]
![image](https://user-images.githubusercontent.com/681355/155465205-ffa18e6f-c2f0-483f-aef5-ee5b666f6f57.png)

I've changed the mapping of "cf" to "w" in settings. Ideally, the help page should reflect the latest mapping. This is very important if you've changed many mappings and have trouble remembering what key maps to what. 

Also if we disable mappings using unmap/unmapAllExcept, these mappings should disappear from the help menu. So we can focus on the shortcuts that are relevant. 

This feature is definitely not necessary but, a nice to have. But having a concise, updated help menu, definitely aids in quick recall and should be prioritised for usability. Having to wade through 100+ shortcuts is too much for most folks. Let's reduce some clutter this way. 

What do you think?

**Comment by b0o** [comment]
I don't have this issue. If I `unmap` a key, it no longer shows up in the help for me. If I `unmapAllExcept("?")`, the only key that shows up is `?`. Can you share your code?

Also keep in mind, if you `map` from one key to another, this creates an additional mapping, it does not change the original one. The help menu will now show both keybindings, not just the new one.

**Comment by b0o** [comment]
Actually, there is one caveat to `unmapAllExcept`: it only unmaps normal-mode mappings. Visual, omnibar, and insert mode mappings will remain. I don't think there is an equivalent `unmapAllExcept` function for those modes.

**Reply by onguarde** [reply]
thanks so much it works. I have the following added in my config, really helps to make the help slimmer. 

```
unmapAllExcept([
    'p','?',';ql','.',      // help
    'cf','gi','gf','[[',']]',';fs','O','f','af','C','i','I','<Ctrl-i>','q','w','Q',         // Mouse click
    'e','d','gg','G',       // scroll page / element
    'yt','yT','g0','g$','gxx','E','R','T',';gt','zr','zi','zo','on','x','X','W','<<','>>',       // Tabs
    'B','F','S','D','r',     // Page navigation
    'sg','sd','se','ss','sh','sy',      // Search selected with
    'yv','yma','ya','yi','yy','yf','yd','cq','cc',';pf',     // Clipboard
    ';e',';v',       // Settings
    'ga','gb','gc','gd','gh','gk','ge','gn','gs',';i',';j',      // Chrome URLs
    ';s',';t',       // Misc
]);
```

Would be great if they have something similar for omnibar and insert mode mappings.

**Reply by b0o** [reply]
Glad to help :) I agree, it seems like those functions are missing. I may submit a PR to add them if I have some time.

**Comment by onguarde** [comment]
Just a small note, if I have this mapping,
`map('w', 'cf');  
`

`unmapAllExcept(['w'])`
I cannot just do this, to hide the default 'cf' mappping, and only show 'w'.

`unmapAllExcept(['w','cf'])`
I need to include BOTH letters. Else neither will show up in the help menu. 

A minor bug I guess, would be great if I can hide 'cf' since I'm not using the default mapping anymore.
